### PR TITLE
Add initial sidecar + ingest cloud function

### PR DESCRIPTION
**Pull request by Fitblip**
Hi friends, this is the block producer stats sidecar that we'll need to deploy for the leaderboard. 

I still need to go get things packaged up, but for now this should allow us to test things out if we want!

### CODE REVIEW

**Comment by bkase** [comment]
Awesome! I just skimmed so far, but can you make sure that this sidecar reconnects to the node if it crashes and restarts? We can also address this by making our node-auto-restart tooling restart this sidecar process too? Maybe that’s better in case this crashes?

**Comment by Fitblip** [comment]
@bkase in the sidecar we already do a graceful crashloop if the node ever goes offline. The initial loop is simply to get the proper "current tip"/finalized block that we should be querying but once we have that the node can go up and down and should only return a proper payload once it's been synced up again.

File: automation/services/mina-bp-stats/sidecar/sidecar.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    assert response.getcode() == 200, "Non-200 from BP flush endpoint! [{}] - ".format(response.getcode(), response.read())
+
+if __name__ == '__main__':
+    logging.info("Starting Mina Block Producer Sidecar")
+    # Go ensure the node is up and happy
+    while True:
+        try:
+            mina_sync_status, head_block_id = fetch_mina_status()
+            if mina_sync_status == "SYNCED" or mina_sync_status == "BOOTSTRAPPING":
```
[the remark this answers is not in the document]
Comment (Fitblip) [reply automation/services/mina-bp-stats/sidecar/sidecar.py:125]: Ah good call!

**Review by bkase — APPROVED** [review]

**Comment by Fitblip** [comment]
Not sure why this got auto-closed when I force pushed my repo 🙄

**Comment by bkase** [comment]
!ci-build-me

**Comment by bkase** [comment]
!approved-for-mainnet

**Comment by bkase** [comment]
(for compatible of course)

**Comment by lk86** [comment]
!ci-build-me

**Review by lk86 — APPROVED** [review]

**Comment by bkase** [comment]
!ci-build-me

**Comment by Fitblip** [comment]
!ci-build-me

**Comment by bkase** [comment]
!ci-build-me

**Comment by lk86** [comment]
!ci-build-me

**Comment by bkase** [comment]
This only adds new code, so it's safe for a hotfix too

**Review by lk86 — APPROVED** [review]

**Comment by lk86** [comment]
!ci-build-me

**Review by 0x0I — APPROVED** [review]

**Pull request merged**